### PR TITLE
ReplicatedMap containsKey should return false on removed keys

### PR DESCRIPTION
**Pull request by ahmetmircik**
closes https://github.com/hazelcast/hazelcast/issues/4420

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/AbstractReplicatedRecordStore.java [hunk 1]
```diff
@@ -257,7 +265,11 @@ public Set entrySet() {
         storage.checkState();
         Set entrySet = new HashSet(storage.size());
         for (Map.Entry<K, ReplicatedRecord<K, V>> entry : storage.entrySet()) {
-            Object key = unmarshallKey(entry.getKey());
+            K keyOfEntry = entry.getKey();
+            if (!containsKey(keyOfEntry)) {
```
Comment (gurbuzali) [inline hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/AbstractReplicatedRecordStore.java]: shouldn't this be containsKeyInternal ?

Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/replicatedmap/impl/record/AbstractReplicatedRecordStore.java]: fixed.

File: hazelcast/src/test/java/com/hazelcast/replicatedmap/ReplicatedMapTest.java [hunk 2]
```diff
@@ -557,6 +559,64 @@ public void testRemoveBinaryDelayDefault() throws Exception {
         testRemove(buildConfig(InMemoryFormat.BINARY, ReplicatedMapConfig.DEFAULT_REPLICATION_DELAY_MILLIS));
     }
 
+    @Test
+    public void testContainsKey_returnsFalse_onRemovedKeys() throws Exception {
+        HazelcastInstance node = createHazelcastInstance();
+        ReplicatedMap<Integer, Integer> map = node.getReplicatedMap("default");
```
Comment (sancar) [inline hazelcast/src/test/java/com/hazelcast/replicatedmap/ReplicatedMapTest.java]: Can we use random map names to resolve potential conflicts with other tests ?

Comment (ahmetmircik) [inline hazelcast/src/test/java/com/hazelcast/replicatedmap/ReplicatedMapTest.java]: There should not be any potential conflict, all instances are using mocked networks.

**Comment by sancar** [comment]
Looks good to me.

**Comment by gurbuzali** [comment]
looks good merging

**Pull request merged**